core/src/main/java/org/apache/carbondata/core/statusmanager/SegmentUpdateStatusManager.java

```diff
@@ -366,17 +366,27 @@ public boolean accept(CarbonFile pathName) {
    */
   public List<String> getDeleteDeltaFilesList(final Segment segment, final String blockName) {
     List<String> deleteDeltaFileList = new ArrayList<>();
-    String segmentPath = CarbonTablePath.getSegmentPath(
-        identifier.getTablePath(), segment.getSegmentNo());
+    String segmentPath = null;
+    for (LoadMetadataDetails details: segmentDetails) {
```

Comment:
check whether the segment is external before this loop like this:

if (segment.isExternalSegment) {
for (LoadMetadataDetails details: segmentDetails) {
...
}
} else {
 segmentPath = CarbonTablePath.getSegmentPath(
              identifier.getTablePath(), segment.getSegmentNo());
}

Reply:
done